src/new-client/src/services/status.test.tsx

```diff
@@ -0,0 +1,148 @@
+      serviceLoad: 1,
+      serviceType: "nlp",
+      timestamp: 1612471950428,
+    },
+  },
+  b: {
+    nlp: {
+      serviceId: "nlp.8895",
+      serviceLoad: 1,
+      serviceType: "nlp",
+      timestamp: 1612471950428,
+    },
+    priceHistory: {
+      serviceId: "priceHistory.a917",
+      serviceLoad: 1,
+      serviceType: "priceHistory",
+      timestamp: 1612471950418,
+    },
+  },
+  c: {
+    nlp: {
+      serviceId: "nlp.8895",
+      serviceLoad: 0,
+      serviceType: "nlp",
+      timestamp: 1612471950428,
+    },
+    priceHistory: {
+      serviceId: "priceHistory.a917",
+      serviceLoad: 1,
+      serviceType: "priceHistory",
+      timestamp: 1612471950418,
+    },
+  },
+}
+
+const wait = (ms: number) => new Promise((res) => setTimeout(res, ms))
+
+const alignResultTimestamp = (
+  mockResult: Record<string, ServiceInstanceStatus>,
+  realResult: Record<string, ServiceInstanceStatus>,
+) => {
+  let currentExpect = mockResult
+  for (const property in mockResult) {
+    currentExpect[property].timestamp = realResult[property].timestamp
+  }
+  return currentExpect
+}
+
+const renderUseStatus = () => {
+  return renderHook(() => useStatus(), {
+    wrapper: ({ children }) => (
+      <Subscribe source$={status$}>{children}</Subscribe>
+    ),
+  })
+}
+
+describe("service/status", () => {
+  describe("useStatus", () => {
+    beforeEach(() => {
+      reset()
+    })
+
+    it("should indicate all the service health when services are connected at beginning", async () => {
+      const mockStream = new BehaviorSubject(mockSource.a)
+      whenWatch("status", mockStream)
+      const { result } = renderUseStatus()
+      expect(result.current).toEqual(
+        alignResultTimestamp(mockResult.a, result.current),
+      )
+
+      await reactAct(async () => {
+        mockStream.next(mockSource.b)
+        await wait(1900)
+      })
+      expect(result.current).toEqual(
+        alignResultTimestamp(mockResult.b, result.current),
+      )
+    })
+
+    it("should show the unhealthy service when one of them are disconnected", async () => {
+      const mockStream = new BehaviorSubject(mockSource.a)
+      whenWatch("status", mockStream)
+      const { result } = renderUseStatus()
+      expect(result.current).toEqual(
+        alignResultTimestamp(mockResult.a, result.current),
+      )
+
+      await reactAct(async () => {
+        await wait(2001)
+        mockStream.next(mockSource.b)
+      })
+      expect(result.current).toEqual(
+        alignResultTimestamp(mockResult.c, result.current),
+      )
+    })
+
+    it("should show the service health again once the service is reconnected", async () => {
+      const mockStream = new BehaviorSubject(mockSource.a)
+      whenWatch("status", mockStream)
+      const { result } = renderUseStatus()
+      expect(result.current).toEqual(
+        alignResultTimestamp(mockResult.a, result.current),
+      )
+
+      await reactAct(async () => {
+        await wait(2001)
```

Comment:
unfortunately this makes tests extremely slow to run, so could you try using [Jest Timer Mocks](https://jestjs.io/docs/en/timer-mocks) instead?

Reply:
sure, I will try